pkg/operator/controller/externaldns/deployment.go

```diff
@@ -411,6 +410,11 @@ func externalDNSContainersChanged(current, expected, updated *appsv1.Deployment)
 				updated.Spec.Template.Spec.Containers[currCont.Index].VolumeMounts = updatedVolumeMounts
 				changed = true
 			}
+			if securityContextChanged(currCont.SecurityContext, expCont.SecurityContext) {
```

Comment:
I'm wondering whether we should reset the whole container's context in case it differs in the fields we want.
The choice is kinda different case by case: for the env vars and args we do hard reset because they may influence the behavior if the externaldns, for the volume mounts we keep the voumes added by Kube.
I think the rule of thumb here is try to not fight for the fields you don't care about. So, following this logic, I'd pass `updated` variable and did the update along the checks.

Reply:
Do you mean, you would prefer to update security context similar to `externalDNSVolumesChanged()`?